modules/memfault-firmware-sdk/memfault_integration.c

```diff
@@ -148,13 +149,57 @@ static int device_info_init(void)
 
 	return err;
 }
-#endif /* CONFIG_MEMFAULT_NCS_DEVICE_ID_IMEI || defined(CONFIG_MEMFAULT_NCS_DEVICE_ID_NET_MAC) */
+#elif defined(CONFIG_MEMFAULT_NCS_DEVICE_ID_IMEI)
+/* Note: IMEI and NET MAC implementations below duplicates functionality in the
+ * hw_id library , but these are deprecated options and will be removed in the
+ * future.
+ */
+static int device_info_init(void)
+{
+	char imei_buf[IMEI_LEN + 6 + 1]; /* Add 6 for \r\nOK\r\n and 1 for \0 */
+
+	/* Retrieve device IMEI from modem. */
+	int err = nrf_modem_at_cmd(imei_buf, ARRAY_SIZE(imei_buf), "AT+CGSN");
+
+	if (err) {
+		LOG_ERR("Failed to get IMEI, error: %d", err);
+		return err;
+	}
+
+	/* Set null character at the end of the device IMEI. */
+	imei_buf[IMEI_LEN] = 0;
+
+	strncpy(device_serial, imei_buf, sizeof(device_serial) - 1);
+
+	return 0;
+}
+#elif defined(CONFIG_MEMFAULT_NCS_DEVICE_ID_NET_MAC)
+static int device_info_init(void)
+{
+	struct net_if *iface = net_if_get_default();
+
+	if (iface == NULL) {
+		return -EIO;
+	}
+
+	struct net_linkaddr *linkaddr = net_if_get_link_addr(iface);
```

Comment:
```suggestion
	struct net_linkaddr *linkaddr = net_if_get_link_addr(iface);
	
	if ((linkaddr == NULL) || (linkaddr->len < 6) {
	        LOG_ERR("Invalid link address");
	        retur -EIO;
	}
```

Reply:
You could also check the addr array size, but that might be a bit too much.

Reply:
good call, will update